src/main/java/com/xebialabs/overthere/cifs/winrs/CifsWinrsConnection.java

```diff
@@ -24,40 +24,37 @@
 
 import com.xebialabs.overthere.*;
 import com.xebialabs.overthere.cifs.CifsConnection;
+import com.xebialabs.overthere.cifs.ConnectionValidator;
 import com.xebialabs.overthere.spi.AddressPortMapper;
-import com.xebialabs.overthere.util.DefaultAddressPortMapper;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import static com.xebialabs.overthere.OperatingSystemFamily.WINDOWS;
 import static com.xebialabs.overthere.cifs.CifsConnectionBuilder.*;
-import static com.xebialabs.overthere.util.OverthereUtils.*;
 import static java.lang.String.format;
 
 /**
  * A connection to a Windows host using CIFS and the Windows native implementation of WinRM, i.e. the <tt>winrs</tt> command.
  */
 public class CifsWinrsConnection extends CifsConnection {
 
-    private ConnectionOptions options;
+    private WinrsConnection connection;
 
-    private OverthereConnection winrsProxyConnection;
+    private ConnectionOptions options;
 
     public CifsWinrsConnection(String type, ConnectionOptions options, AddressPortMapper mapper) {
         super(type, options, mapper, true);
-        checkArgument(os == WINDOWS, "Cannot create a " + CIFS_PROTOCOL + ":%s connection to a machine that is not running Windows", cifsConnectionType.toString().toLowerCase());
-        checkArgument(mapper instanceof DefaultAddressPortMapper, "Cannot create a " + CIFS_PROTOCOL + ":%s connection when connecting through a SSH jumpstation", cifsConnectionType.toString().toLowerCase());
-        checkArgument(password.indexOf('\'') == -1 && password.indexOf('\"') == -1, "Cannot create a " + CIFS_PROTOCOL + ":%s connection with a password that contains a single quote (\') or a double quote (\")", cifsConnectionType.toString().toLowerCase());
-
+        ConnectionValidator.assertIsWindowsHost(os, CIFS_PROTOCOL, cifsConnectionType);
+        ConnectionValidator.assetNotThroughJumpstation(mapper, CIFS_PROTOCOL, cifsConnectionType);
+        ConnectionValidator.assertNoSingleQuoteInPassword(password, CIFS_PROTOCOL, cifsConnectionType);
         this.options = options;
     }
 
     @Override
     public void connect() {
-        connectToWinrsProxy(options);
+        connection = new WinrsConnection(options, mapper, workingDirectory);
+        connection.connectToWinrsProxy(options);
 
-        if (winrsProxyConnection.getHostOperatingSystem() != WINDOWS) {
-            disconnectFromWinrsProxy();
+        if (connection.getWinrsProxyConnection().getHostOperatingSystem() != WINDOWS) {

```

Comment:
Please see
https://github.com/xebialabs/overthere#cifs_winrsProxyProtocol
In constructor validation we check that the OS selected on CI is windows, subsequent to that we create a local connection check that OS is windows, this check internally uses System.getProperty("os.name") to getOS information